src/Serilog.Sinks.AzureTableStorage/LoggerConfigurationAzureTableStorageExtensions.cs

```diff
@@ -74,13 +75,174 @@ public static class LoggerConfigurationAzureTableStorageExtensions
+                return AzureTableStorage(loggerConfiguration, null, storageAccount, restrictedToMinimumLevel, formatProvider, storageTableName, writeInBatches, period, batchPostingLimit, keyGenerator, true, cloudTableProvider);
+            }
+            catch (Exception ex)
+            {
+                Debugging.SelfLog.WriteLine($"Error configuring AzureTableStorage: {ex}");
+
+                ILogEventSink sink = new LoggerConfiguration().CreateLogger();
+                return loggerConfiguration.Sink(sink, restrictedToMinimumLevel);
+            }
+        }
+
+
+        /// <summary>
+        /// Adds a sink that writes log events as records in an Azure Table Storage table (default LogEventEntity) using the given storage account.
+        /// </summary>
+        /// <param name="loggerConfiguration">The logger configuration.</param>
+        /// <param name="textFormatter">Use a Serilog ITextFormatter such as CompactJsonFormatter to store object in data column of Azure table</param>
+        /// <param name="storageAccount">The Cloud Storage Account to use to insert the log entries to.</param>
+        /// <param name="restrictedToMinimumLevel">The minimum log event level required in order to write an event to the sink.</param>
+        /// <param name="formatProvider">Supplies culture-specific formatting information, or null.</param>
+        /// <param name="storageTableName">Table name that log entries will be written to. Note: Optional, setting this may impact performance</param>
+        /// <param name="writeInBatches">Use a periodic batching sink, as opposed to a synchronous one-at-a-time sink; this alters the partition
+        /// key used for the events so is not enabled by default.</param>
+        /// <param name="batchPostingLimit">The maximum number of events to post in a single batch.</param>
+        /// <param name="period">The time to wait between checking for event batches.</param>
+        /// <param name="keyGenerator">The key generator used to create the PartitionKey and the RowKey for each log entry</param>
+        /// <param name="bypassTableCreationValidation">Bypass the exception in case the table creation fails.</param>
+        /// <param name="cloudTableProvider">Cloud table provider to get current log table.</param>
+        /// <returns>Logger configuration, allowing configuration to continue.</returns>
+        /// <exception cref="ArgumentNullException">A required parameter is null.</exception>
+        public static LoggerConfiguration AzureTableStorage(
+            this LoggerSinkConfiguration loggerConfiguration,
+            ITextFormatter textFormatter,
+            CloudStorageAccount storageAccount,            
+            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
+            IFormatProvider formatProvider = null,
```

Comment:
An `IFormatProvider` shouldn't be required when an `ITextFormatter` is specified.